pkg/apis/management.cattle.io/v3/public_authn_types.go

```diff
@@ -174,3 +174,24 @@ type SamlLoginInput struct {
 type SamlLoginOutput struct {
 	IdpRedirectURL string `json:"idpRedirectUrl"`
 }
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type OIDCProvider struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	AuthProvider      `json:",inline"`
+
+	RedirectURL string `json:"redirectUrl"`
+}
+
+type OIDCLogin struct {
+	GenericLogin `json:",inline"`
+	Code         string `json:"code" norman:"type=string,required"`
+}
+
+type KeyCloakOIDCProvider struct {
+	OIDCProvider `json:",inline"`
+}
```

Comment:
Do we need KeyCloak Login ?

Reply:
The keycloak configuration is going to be using the OIDCLogin struct for it so we shouldn't need to make more structs just for keycloak